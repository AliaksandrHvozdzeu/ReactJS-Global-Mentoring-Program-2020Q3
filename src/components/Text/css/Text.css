.text {
    color: var(--white);
    font-size: 20px;
}
