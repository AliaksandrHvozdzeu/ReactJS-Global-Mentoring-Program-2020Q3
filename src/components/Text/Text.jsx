import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

export default function Text({ id, title, value }) {

  return (
    <>
      <label htmlFor={id}>
        {title}
        <p id={id}>{value}</p>
      </label>
    </>
  );

}

Text.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

Text.defaultProps = {
  value: '',
};
