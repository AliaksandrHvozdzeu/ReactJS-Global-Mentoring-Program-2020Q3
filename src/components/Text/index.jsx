import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const Text = ({ id, title, value }) => (
  <label htmlFor={id}>
    {title}
    <p id={id}>{value}</p>
  </label>
);

Text.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

Text.defaultProps = {
  value: ''
};

export default Text;
