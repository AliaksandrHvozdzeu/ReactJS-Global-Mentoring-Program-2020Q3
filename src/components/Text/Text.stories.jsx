import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import Text from './js/Text';
import './css/Text.css';
import '../../App.css';

export default {
  title: 'Text',
  decorators: [withKnobs],
};

export const Default = () => (
  <>
    <label htmlFor="text_id">
      "Text header"
      <p id="text_id" className="text">Text value</p>
    </label>
  </>
);
