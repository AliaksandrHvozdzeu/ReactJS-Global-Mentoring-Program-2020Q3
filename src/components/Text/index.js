import Text from './js/Text';

export default Text;
