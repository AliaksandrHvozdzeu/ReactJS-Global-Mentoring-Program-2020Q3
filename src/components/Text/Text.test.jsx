import React from 'react';
import { render } from 'enzyme';
import rerender from 'react-test-renderer';
import Text from './js/Text';

describe('when Text', () => {
  test('then snapshot created', () => {
    const TextComponentId = 'test-text-id';
    const TextComponentTitle = 'Test text title';
    const TextComponentValue = 'Test text value';
    const component = rerender.create(<Text id={TextComponentId}
                                            title={TextComponentTitle}
                                            value={TextComponentValue}/>);
    const textComponent = render(<Text id={TextComponentId}
                                       title={TextComponentTitle}
                                       value={TextComponentValue}/>);
    expect(textComponent.find('.text').text()).toEqual('Test text value');
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
