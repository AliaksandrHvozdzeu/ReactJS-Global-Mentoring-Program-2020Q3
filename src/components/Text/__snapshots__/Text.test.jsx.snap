// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Text then snapshot created 1`] = `
<label
  htmlFor="test-text-id"
>
  Test text title
  <p
    className="text"
    id="test-text-id"
  >
    Test text value
  </p>
</label>
`;
